db/compaction/compaction.cc

```diff
@@ -664,13 +664,6 @@ bool Compaction::ShouldFormSubcompactions() const {
     return false;
   }
 
-  // Note: the subcompaction boundary picking logic does not currently guarantee
-  // that all user keys that differ only by timestamp get processed by the same
-  // subcompaction.
-  if (cfd_->user_comparator()->timestamp_size() > 0) {
```

Comment:
Once all the tests passes and PR is accepted I will add this check again and handle it in next PR